src/TuxxAnimations.js

```diff
@@ -49,59 +50,323 @@ var createAnimationClass = function (transitions, customClassName) {
+            }
+            enter['transition-duration'] = duration;
+            enterActive['transition-duration'] = duration;
+            leave['transition-duration'] = duration;
+            leaveActive['transition-duration'] = duration;
+          }
+          if (enterProps) {
+            if (typeof enterProps.duration === "number") {
+              enterProps.duration = enterProps.duration + "ms";
             }
+            enter['transition-duration'] = enterProps.duration;
+            enterActive['transition-duration'] = enterProps.duration;
+          }
+          if (leaveProps) {
+            if (typeof leaveProps.duration === "number") {
+              leaveProps.duration = leaveProps.duration + "ms";
+            }
+            leave['transition-duration'] = leaveProps.duration;
+            leaveActive['transition-duration'] = leaveProps.duration;
+          }
 
-            //Easing prop work-------------------------------------------
-            var easing = this.props.easing;
-            //Use default if no easing prop defined
-            if (easing) {
-              //Check to see if easing exists in default Easings object from the AnimationEasings module
-              if (easing in Easings) {
-                //Set easing to value of correpsonding key from Easings module object
-                easing = Easings[easing];
-              }
-              transition['transition-timing-function'] =  easing;
+          //easing prop work-------------------------------------------
+          var easing = this.props.easing;
+          //use default if no easing prop defined
+          if (easing) {
+            //check to see if easing exists in default Easings object from the AnimationEasings module
+            if (Easings[easing]) {
+              //set easing to value of correpsonding key from Easings module object
+              easing = Easings[easing];
+              enter['transition-timing-function'] = easing;
+              enterActive['transition-timing-function'] = easing;
+              leave['transition-timing-function'] = easing;
+              leaveActive['transition-timing-function'] = easing;
+            } else {
+              throw 'easing: not of acceptable type. Check the TuxxAnimationEasings file for a list of accepted easings or use a cubic bezier input.'
+            }
+          }
+          if (enterProps) {
+            //check to see if easing exists in default Easings object from the AnimationEasings module
+            if (Easings[enterProps.easing]) {
+              //set easing to value of correpsonding key from Easings module object
+              enter['transition-timing-function'] =  Easings[enterProps.easing];
+              enterActive['transition-timing-function'] = Easings[enterProps.easing];
+            } else {
+              throw 'easing: not of acceptable type. Check the TuxxAnimationEasings file for a list of accepted easings or use a cubic bezier input.'
```

Comment:
Instead of throwing errors, use invariant. Look at other files in the tuxx/src for examples.